### PR TITLE
Adding CiC compensation + Online PGD fitting

### PR DESCRIPTION
**Pull request by EiffL**
This PR adds the following marginal improvements to PGD:

- Include compensation kernel for CiC window in PGD loss function
- Small rename of pgd functions for consistency
- Adding online PGD capability + training script

This is meant to solve #89

### CODE REVIEW

**Review by dlanzieri — APPROVED** [review]
Thank you @EiffL! So, can we consider also close the problem related to the  weight function?

**Review by dlanzieri — APPROVED** [review]

**Comment by EiffL** [comment]
yes, no weight function anymore
